asllib/tests/asl/required/subtypes-with.asl

```diff
@@ -0,0 +1,39 @@
+type rec1 of record {
+  a: integer,
+  b: integer,
+};
+
+type rec2 subtypes rec1 with { 
+  c: integer,
+};
+
+type rec2bis subtypes rec1 with { 
+  c: integer,
+};
+
+func main() => integer
+begin
+  let a = rec2 {
+    a = 1,
+    b = 2,
+    c = 3
```

Comment:
Optional ',' after the last field definition is _not_ allowed. Perhaps we could allow this optional comma everywhere? Notice that the ASL document does not allow those terminal ','.

Reply:
Well thinking twice, following the official grammar is the best option. If we do not we encourage syntax that other tools may not accept. So the current parser is fine.

This does not mean not lobbying for a more consistent syntax,. A minor issue though.

Reply:
Good catch! I somehow managed not to commit it...

It should be fixed with the last pushed version.

I also integrated your example, see the error message in `asllib/tests/asl.t/run.t`. Do you think I should test more on this feature?